tsdb/series_file.go

```diff
@@ -35,16 +37,30 @@ type SeriesFile struct {
 	path       string
 	partitions []*SeriesPartition
 
+	maxCompactionConcurrency int
+
 	refs sync.RWMutex // RWMutex to track references to the SeriesFile that are in use.
 
 	Logger *zap.Logger
 }
 
 // NewSeriesFile returns a new instance of SeriesFile.
-func NewSeriesFile(path string) *SeriesFile {
+func NewSeriesFile(path string, maxCompactionConcurrency int) *SeriesFile {
```

Comment:
I think that `maxCompactionConcurrency` should/could have been a functional option. Then there would not have needed to be any API changes to existing code in tests etc.

Something like:

```go
type SeriesFileOption func(*SeriesFile)

var WithCompactionConcurrency = func (s *SeriesFile) {
    // maxCompactionConcurrency logic....
}
```